pkg/storage/index.go

```diff
@@ -40,6 +51,116 @@ func GetRecordIndex(msg proto.Message) *structpb.Struct {
 	return f.GetStructValue()
 }
 
+func handleAny(val any) (string, error) {
+	switch v := val.(type) {
+	case string:
+		return v, nil
+	default:
+		return "", fmt.Errorf("%T : %w", v, ErrIndexUnsupportedProtoField)
+	}
+}
+
+func handleMap(in map[string]any, fieldTable map[string]string, fields []string) error {
+	for _, key := range fields {
+		parts := strings.Split(key, ".")
+		cur := in
+		for partIdx, part := range parts {
+			val, ok := cur[part]
+			if !ok {
+				break
+			}
+			if partIdx == len(parts)-1 {
+				indexVal, err := handleAny(val)
+				if err != nil {
+					return err
+				}
+				fieldTable[key] = indexVal
+			} else {
+				next, ok := val.(map[string]any)
+				if !ok {
+					break
+				}
+				cur = next
+			}
+		}
+	}
+	return nil
+}
+
+func GetIndexableFields(msg *anypb.Any, fields []string) (mappings map[string]string, err error) {
+	fieldTable := slices.Associate(fields, func(field string) (string, string) {
+		return field, ""
+	})
+	mapSt, err := anyToMap(msg)
+	if err != nil {
+		return nil, err
+	}
+
+	err = handleMap(mapSt, fieldTable, fields)
```

Comment:
@kralicky  @calebdoxsey 

This is not a great way to deref fields on protobuf messages encoded into an *anypb.Any. Maybe there is a better/more efficient way? I couldn't find any in the built-in packages

Reply:
Not sure it's any better, but we have some similar code in pkg/logutil/scrub.go